test/client/session.lua

```diff
@@ -202,14 +227,21 @@ function Session:_run(request_cb, notification_cb, timeout)
       self._prepare:stop()
     end)
   end
-  self._msgpack_rpc_stream:read_start(request_cb, notification_cb, function()
+  self._rpc_stream:read_start(request_cb, notification_cb, function()
     uv.stop()
-    self.eof_err = { 1, 'EOF was received from Nvim. Likely the Nvim process crashed.' }
+
+    --- @diagnostic disable-next-line: invisible
+    local stderr = self._rpc_stream._stream.stderr --[[@as string?]]
+    -- See if `proc.stderr` has anything useful.
+    stderr = '' ~= ((stderr or ''):match('^%s*(.*%S)') or '') and ' stderr:\n' .. stderr or ''
+
+    self.eof_err = { 1, 'EOF was received from Nvim. Likely the Nvim process crashed.' .. stderr }
```

Comment:
Include stderr (if any) in the EOF message. Not sure if this will ever be useful, will see...